src/EA.Weee.Web/Areas/Admin/Controllers/ProducerSubmissionController.cs

```diff
@@ -129,10 +137,67 @@ public async Task<ActionResult> AddPaymentDetails()
             return View();
         }
 
+        [AdminSmallProducerSubmissionContext]
         [HttpGet]
-        public async Task<ActionResult> RemoveSubmission()
+        public async Task<ActionResult> RemoveSubmission(Guid registeredProducerId, string registrationNumber, int year)
         {
-            return View();
+            var selectedValue = string.Empty;
+            var model = new ConfirmRemovalViewModel
+            {
+                SelectedValue = selectedValue,
+                Producer = new ProducerDetailsScheme
+                {
+                    ComplianceYear = year,
+                    ProducerName = this.SmallProducerSubmissionData.OrganisationData.OrganisationName,
+                    RegistrationNumber = registrationNumber,
+                    RegisteredProducerId = registeredProducerId
+                }
+            };
+
+            return View(model);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> RemoveSubmission(ConfirmRemovalViewModel viewModel)
```

Comment:
@stevelee-civica  Sean is making a change to only enable the remove button for admin so on these attributes we should have  [AuthorizeInternalClaims(Claims.InternalAdmin)]

Reply:
updated